R/drive_cp.R

```diff
@@ -47,35 +49,39 @@ drive_cp <- function(file, path = NULL, name = NULL, verbose = TRUE) {
   }
 
   if (is_path(path)) {
-    if (is.null(name) && drive_path_exists(append_slash(path))) {
-      path <- append_slash(path)
-    }
+    confirm_clear_path(path, name)
     path_parts <- partition_path(path, maybe_name = is.null(name))
     path <- path_parts$parent
     name <- name %||% path_parts$name
   }
 
-  name <- name %||% glue("Copy of {file$name}")
+  name <- name %||% glue::glue("Copy of {file$name}")
 
   params <- list(
     fileId = file$id,
     fields = "*"
   )
 
   ## if copying to a specific directory, specify the parent
-  if (!is.null(path)) {
-    path <- as_dribble(path)
-    confirm_single_file(path)
-    if (!is_folder(path)) {
-      stop(
-        glue(
-          "Requested parent folder does not exist:\n{path$name}"
-        ),
-        call. = FALSE
-      )
-    }
-    params[["parents"]] <- list(path$id)
+  ## defaults to the current file's directory
+  path <- path %||% as_id(file$files_resource[[1]]$parents[[1]])
+  path <- as_dribble(path)
+  if (!some_files(path)) {
+    sglue("Requested parent folder does not exist.")
   }
+  if (!single_file(path)) {
+    paths <- glue::glue_data(path, "  * {name}: {id}")
+    scollapse(
+      c("Requested parent folder identifies multiple files:", paths),
+      sep = "\n"
+    )
+  }
+  ## if path was input as a dribble or id, still need to be sure it's a folder
+  if (!is_folder(path)) {
+    sglue("\n`path` specifies a file that is not a folder:\n * {path$name}")
+  }
+  params[["parents"]] <- list(path$id)
+
```

Comment:
I'm not sure about this (the copying of the parent). Can you check against Drive operations in the browser? If file A has folder B as parent and you make a copy of A, does the copy have folder B as parent?

It feels like we should either stay out of this or do some global treatment of metadata. But mostly it feels like something we shouldn't get into. If we do it, what about multiple parents? Because that doesn't seem accounted for.